src/Illuminate/Queue/Console/WorkCommand.php

```diff
@@ -91,6 +91,17 @@ public function fire()
      */
     protected function runWorker($connection, $queue, $delay, $memory, $daemon = false)
     {
+        $timeout = $this->option('timeout');
```

Comment:
we should  throw an exception is timeout is asked for, but `pcntl_signal` is disabled